### PR TITLE
Update KafkaConfig.scala

### PR DESCRIPTION
**Pull request by gemma-singleton**
Better clarify the auto leader rebalance config documentation

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by stanislavkozlovski — APPROVED** [review]

**Review by stanislavkozlovski — APPROVED** [review]
Thanks for the PR, LGTM!

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 1]
```diff
@@ -688,7 +688,7 @@ object KafkaConfig {
   val FetchPurgatoryPurgeIntervalRequestsDoc = "The purge interval (in number of requests) of the fetch request purgatory"
   val ProducerPurgatoryPurgeIntervalRequestsDoc = "The purge interval (in number of requests) of the producer request purgatory"
   val DeleteRecordsPurgatoryPurgeIntervalRequestsDoc = "The purge interval (in number of requests) of the delete records request purgatory"
-  val AutoLeaderRebalanceEnableDoc = "Enables auto leader balancing. A background thread checks and triggers leader balance if required at regular intervals"
+  val AutoLeaderRebalanceEnableDoc = "Enables auto leader balancing. A background thread checks the distribution of partition leaders at regular intervals, configurable by `leader.imbalance.check.interval.seconds`. If required, it triggers a leader rebalance to the preferred leader for partitions"
```
Comment (kamalcph) [inline core/src/main/scala/kafka/server/KafkaConfig.scala]: May be we can also include the config `leader.imbalance.per.broker.percentage` in the description.

If it exceeds the `leader.imbalance.per.broker.percentage`, it triggers a leader...

Comment (gemma-singleton) [reply core/src/main/scala/kafka/server/KafkaConfig.scala]: Yes I think that makes sense, I'll look at re-writing that into the description today

**Review by junrao — APPROVED** [review]
@gemma-singleton : Thanks for the PR. LGTM

**Pull request merged**